lib/quoil/cli.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Quoil.CLI do
+  @moduledoc """
+  Command line interface for the quoil function  
+    -> parsing the arguments
+    -> running the ping command  
+    -> interpreting the results  
+    -> logging
+    """
+
+  def main(argv) do
+    argv
+    |> Quoil.ArgsProcessor.parse_args
+    |> terminate_early?
+    |> run_ping
+    |> Quoil.ParseResults.parse_result
+    |> Quoil.LogResults.write_log
+    |> Quoil.Repeat.requested?
+
+  end
+
+
+  def terminate_early?(:help) do
+    IO.puts """
+    usage: quoil [-h | --help]
+    quoil [--interval sec] [--number nr] [--repeat nr [--wait min]] <ip_to_ping> [log_file_name]
+    
+    all values supplied must be positive integers
+
+    For more information see: https://github.com/rawdamedia/quoil
+    """
+    System.halt(0)
+  end
+  def terminate_early?(:error) do
+    IO.puts "ERROR: There was an error with the command line switches."
+    System.halt(:abort)
+  end
+  def terminate_early?(switches) do
+    switches
+  end
+  
+
+  @doc"""
+  Runs the System `ping` command.  
+  Uses *ip_to_ping* and *switches* to appropriately format the `ping` command-line options.  
+  Returns a tuple `{result, switches, log_file_name}` where:
+  - *result* is the raw result of running the `ping` command.
+  - *switches* is the unmodified map containing all the switches.
+  - *log_file_name* is passed along unmodified.
+  """
+  def run_ping({ip_to_ping, switches, log_file_name}) do
+    args = ["-q", "-c", to_string(switches[:number]), "-W", to_string(switches[:interval]), ip_to_ping]
+    {rslt, exit_code} = System.cmd("ping", args)
```

Comment:
Unfortunately my concern has been confirmed (at least on my mac)

In order to get gen_icmp installed properly I had to run `sudo mix deps.compile` as just doing `mix deps.get` did not generate the dynamic library for _procket_.  

Opening up a socket seems to work in non-privileged mode (a PID is returned) but sending pings through it returns:
> [{:error, :timeout, 'telstra.com.au', {58, 162, 89, 137}}, {:error, :timeout, 'www.google.com.au', {58, 162, 62, 84}}]

If this is run via `sudo` it works properly:
> [{:ok, 'telstra.com.au', {58, 162, 89, 137}, {58, 162, 89, 137}, {49914, 1, 243, 72}, " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJK"}, {:ok, 'www.google.com.au', {58, 162, 62, 84}, {58, 162, 62, 84}, {49914, 0, 60, 65}, " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJK"}]

... hold that thought ...

if I follow the _procket_ readme.md instructions to setuid by running:

`sudo chmod 4750 deps/procket/priv/procket`

then executing as non privileged user seems to work OK.  Now all I have to figure out is:
- does escript package up the procket functionality with appropriate permissions set?
- is it possible to change the settings in my app so that when the repo is cloned, running `mix deps.get` asks for superuser permission prior to trying to compile the dependencies?
- is it possible to have `mix deps.get` or `mix deps.compile` run a `chmod` to allow for setuid on procket?